scripts/export-resources.sh

```diff
@@ -6,6 +6,49 @@
 # arg2: target file path, such as "/tmp/my_application.yaml"
 # arg3: input "with-cluster" means export all dockerregistries, deploykeys, httpscerts.
 
+function export_resource () {
+  ns=$1
+  crd=$2
+  item=$3
+
+  cmd="kubectl get "
+  patch_cmd="kubectl patch "
+  if [ "$1" = "-" ]; then
```

Comment:
It seems if the parameters are different the export resource function will have significant different behaviors. Why not use two function?

Reply:
Just to reuse the code, the logic here looks a bit complicated, it could be split into three methods.